transformer/external/executabletransformer.go

```diff
@@ -32,20 +32,48 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	envMode      = "env"
+	argMode      = "arg"
+	envDelimiter = "="
+)
+
 // Executable implements transformer interface and is used to write simple external transformers
 type Executable struct {
 	Config     transformertypes.Transformer
 	Env        *environment.Environment
 	ExecConfig *ExecutableYamlConfig
 }
 
+// InputFormat input/output mode definition
+type InputFormat struct {
+	Mode    string `yaml:"mode"`
+	EnvName string `yaml:"envName,omitempty`
+	Path    string
+}
+
+// OutputFormat input/output mode definition
+type OutputFormat struct {
+	Mode    string `yaml:"mode"`
+	EnvName string `yaml:"envName,omitempty`
+	Path    string
+}
+
+// InputOutput input/output mode definitions for detect script
+type InputOutput struct {
+	Input  InputFormat  `yaml:"input"`
+	Output OutputFormat `yaml:"output"`
+}
+
 // ExecutableYamlConfig is the format of executable yaml config
 type ExecutableYamlConfig struct {
 	EnableQA           bool                       `yaml:"enableQA"`
 	Platforms          []string                   `yaml:"platforms"`
 	DirectoryDetectCMD environmenttypes.Command   `yaml:"directoryDetectCMD"`
 	TransformCMD       environmenttypes.Command   `yaml:"transformCMD"`
 	Container          environmenttypes.Container `yaml:"container,omitempty"`
+	DetectScriptIO     InputOutput                `yaml:"detectScriptIO"`
+	TransformScriptIO  InputOutput                `yaml:"transformScriptIO"`
```

Comment:
Do we need separate variables for this? Wouldn't keeping this with the CMD or the environment like `container` make it more coherent?

Reply:
The rationale behind have separate variables is to make it independent of the environment type (container, local etc). Also, 
(1) it allows the added flexibility of defining the I/O for detect and transform scripts separately. 
(2) We could potentially use the same format for the upcoming gRPC invocation method.

Reply:
done